crates/cairo-lang-utils/src/bigint.rs

```diff
@@ -112,44 +112,40 @@ where
 
 impl Encode for BigIntAsHex {
     fn size_hint(&self) -> usize {
-        // sign, len, data.
-        1 + 8 + self.value.to_bytes_be().1.len()
+        // sign (2 is the biggest sign value possible), data.
+        (2u8, self.value.to_bytes_le().1).size_hint()
```

Comment:
What value does this return?
I expect it to be equal to `1 + 1 + self.value.to_bytes_le().1.len()`

Reply:
I don't like this implementation uses `to_bytes_le` which is performing memory allocation internally (a new vec is crated).

I think `size_hint` should not do that. Just read the data and compute it's size.
YOu can instead use `mem::size_of_val`. That will look at the size of the inner vec

Reply:
`it prunes the 0 bytes` what do you mean?

Reply:
You can get the inner `data` by calling the `IntDigits` trait `digits` method.
And actually, it's true length. Then you just have to multiply by 8

```rust
impl IntDigits for BigUint {
    #[inline]
    fn digits(&self) -> &[BigDigit] {
        &self.data
    }
    #[inline]
    fn digits_mut(&mut self) -> &mut Vec<BigDigit> {
        &mut self.data
    }
    #[inline]
    fn normalize(&mut self) {
        self.normalize();
    }
    #[inline]
    fn capacity(&self) -> usize {
        self.data.capacity()
    }
    #[inline]
    fn len(&self) -> usize {
        self.data.len()
    }
}

```

Reply:
This is the implementation of `to_bytes_le()`
```rs
/// Returns the byte representation of the `BigUint` in little-endian byte order.
    ///
    /// # Examples
    ///
    /// ```
    /// use num_bigint::BigUint;
    ///
    /// let i = BigUint::parse_bytes(b"1125", 10).unwrap();
    /// assert_eq!(i.to_bytes_le(), vec![101, 4]);
    /// ```
    #[inline]
    pub fn to_bytes_le(&self) -> Vec<u8> {
        if self.is_zero() {
            vec![0]
        } else {
            convert::to_bitwise_digits_le(self, 8)
        }
    }
```
If i used `core::mem::size_of(val)` it would have returned 8 which would be wrong